robyn/openapi.py

```diff
@@ -124,6 +124,45 @@ class OpenAPIInfo:
     components: Components = field(default_factory=Components)
 
 
+@dataclass
+class MediaType:
+    """
+    Each Media Type Object provides schema and examples for the media type identified by its key.
+    (https://swagger.io/specification/#media-type-object)
+
+    @param schema: Optional[Dict] The schema defining the content of the request, response, or parameter. For more
+    information about the properties, see
+    <a href="https://tools.ietf.org/html/draft-bhutton-json-schema-00">JSON Schema Core</a>
+    and <a href="https://tools.ietf.org/html/draft-bhutton-json-schema-validation-00">JSON Schema Validation</a>.
+    @param example: Optional[Any] Example of the media type. The example object SHOULD be in the correct format
+    as specified by the media type.
+    @param examples: Optional[Dict[str, dict]] Examples of the media type. Each example object SHOULD match the media
+    type and specified schema if present.
+    @param encoding: Optional[Dict[str, dict]] A map between a property name and its encoding information.
+    """
+
+    schema: Optional[Dict] = field(default_factory=dict)
```

Comment:
the openapi `schema` supports all the properties from the [JSON Schema Specification Draft](https://tools.ietf.org/html/draft-bhutton-json-schema-00). ref: [https://swagger.io/specification/#schema-object](https://swagger.io/specification/#schema-object). I don't see a point in us creating a class for this; for we would restrict the user from using all the params. what do you think?

Reply:
@VishnuSanal , we only care about the user's developer experience. It is foolish of us to expect the user to write openapi style json for documentation.

If we want the user to write openpai JSON for the body, why even bother implementing this feature in the first place?

I hope this answers your question.

Reply:
> we only care about the user's developer experience. It is foolish of us to expect the user to write openapi style json for documentation.

I can agree; but, my concern was whether we'll restrict the user from using any of the params mentioned in the entirety of the JSON schema draft that the OAS schema key supports.

Rephrasing myself: say, we implement a data class for schema with the most used keys. What if a user wants to use a key that we haven't included? Or, if we choose to add support for **all keys** in the entire schema draft, will it be worth the effort we put in to support all of those keys inside a data class -- for we don't know whether they'll be used or not.

I was opting for flexibility for the user to choose which keys that they want to use by letting the user input the schema as a dict (like we did for all the keys inside the components object).

Please correct me if my assumptions are wrong, or if there's a better way around to get the best of both worlds. Thanks!

> If we want the user to write openpai JSON for the body, why even bother implementing this feature in the first place?
>
> I hope this answers your question.

I am not sure why you are being furious about this.

Reply:
> I am not sure why you are being furious about this.

There are two reasons. I'll explain in tomorrow's meeting